src/app/meteo/form-filter/form-filter.component.html

```diff
@@ -0,0 +1,114 @@
+
+            <div class="form-group row"
+                 [ngClass]="{'has-error': filterForm.get('rows').dirty && filterForm.get('rows').errors,
+                             'has-success': filterForm.get('rows').dirty && !filterForm.get('rows').errors}">
+                <div class="col-xs-8">
+                    <label for="input-rows">Row of table*:</label>
+                    <select class="form-control"
+                            id="input-rows"
+                            formControlName="rows">
+                        <option [selected]="filterForm.get('rows').value === null" value="">-- Select --</option>
+                        <option *ngFor="let row of rowsList" [value]="row">{{row}}</option>
+                    </select>
+                    <wapi-input-errors [inputErrors]="filterForm.get('rows').dirty && filterForm.get('rows').errors"
+                                       [inputValid]="'Valid. Nice choice.'"
+                                       [errorsState]="{required: 'Choose visible number of rows'}"></wapi-input-errors>
+                </div>
+            </div>
+
+            <div class="form-group">
+                <label>Measure:</label> <br/>
+                <label class="radio-inline" *ngFor="let unit of measureList">
+                    <input type="radio"
+                           [value]="unit"
+                           formControlName="measure">
+                    {{unit}}
+                </label>
+            </div>
+
+            <div class="form-group">
+                <label>Hide table head:</label> <br/>
+                <wapi-checkbox formControlName="toggle"></wapi-checkbox>
+            </div>
+
+            <hr/>
+
+            <div class="form-group row"
+                 [ngClass]="{'has-error': filterForm.get('cityName').dirty && filterForm.get('cityName').errors,
+                             'has-success': filterForm.get('cityName').dirty && !filterForm.get('cityName').errors}">
+                <div class="col-xs-8">
+                    <label for="input-city">Get weather:</label>
+                    <input type="text"
+                           class="form-control"
+                           placeholder="Enter city"
+                           formControlName="cityName"
+                           id="input-city">
+                    <wapi-input-errors [inputErrors]="filterForm.get('cityName').dirty && filterForm.get('cityName').errors"
+                                       [inputValid]="'Now it\'s a valid'"
+                                       [errorsState]="{minlength: 'City name should consist at least 3 letters!'}"></wapi-input-errors>
+                </div>
+                <div class="col-xs-4">
+                    <label>&nbsp;</label>
+                    <button type="button"
+                            class="btn btn-primary btn-block"
+                            [disabled]="!filterForm.get('cityName').dirty || filterForm.get('cityName').errors"
```

Comment:
@LeraKruchek no. What I've meant, I'd written. If input cityName is touched (not dirty) OR cityName has errors. In your case cta-button enabled if cityName is empty.